mongodbatlas/provider_test.go

```diff
@@ -249,3 +255,19 @@ func testCheckPrivateEndpointServiceDataFederationOnlineArchiveRun(tb testing.TB
 		tb.Skip("`MONGODB_ATLAS_PRIVATE_ENDPOINT_ID` must be set for Private Endpoint Service Data Federation and Online Archive acceptance testing")
 	}
 }
+
+var _ plancheck.PlanCheck = debugPlan{}
+
+type debugPlan struct{}
+
+func (e debugPlan) CheckPlan(ctx context.Context, req plancheck.CheckPlanRequest, resp *plancheck.CheckPlanResponse) {
```

Comment:
where is this function used?

Reply:
This method is essentially used to ensure that the generated plan is empty, it is the same that we are using for migration to the Plugin Framework as well. 

I have added the same migration tests here that we use for migrating to the Plugin Framework as is, as the tests essentially help us identify if we might be introducing any unwanted changes to existing implementations of the resources.